### ISSUE TITLE
Marketplace mode instalation issue

### ISSUE THREAD

**Issue by Wolvverine** [cconard96/glpi-camerainput-plugin#8]
in Marketplace mode plugin do not show problem with instalation:
"This plugin requires be served over HTTPS"

**Comment by ashledombos** [comment]
Same with standard plugin installation.

**Comment by cconard96** [comment]
Are you connected to GLPI over HTTPS rather than plain HTTP?
Browsers won't let web apps access your camera unless it is over a secure connection so there is an explicit check during install to make sure you have your web server set up right.

If you are connected over HTTPS, is your GLPI behind a proxy?

**Comment by ashledombos** [comment]
It is HTTPS.

Indeed, it is behind a reverse-proxy. To be more precise, it is a docker stack: reverse-proxy (ssl) > web > php

**Comment by ashledombos** [comment]
I also have the same issue with https://github.com/cconard96/glpi-screenshot-plugin, so I guess somehow these plugins don’t see that GLPI is accessed through SSL.

**Comment by cconard96** [comment]
Most likely... both plugins use browser features that only work over secure connections.
The only thing I'm doing to check HTTPS is making sure $_SERVER["HTTPS"] is true or the source IP is localhost. This should be set by PHP interpreter if the request is made over HTTPS.

I haven't been able to understand yet why this is broken for everyone that has GLPI behind a reverse proxy though.

Is the dockerfile public for your stack? If not, could you give some details about the docker environment so I can try to recreate it on my end?

**Comment by ashledombos** [comment]
Sure

I have no problem to share my whole stack, but I’m afraid I would give too much information. But I guess the issue could come from the reverse proxy side (here Traefik)

docker-compose.yml
```
version: '2.5'

services:
  traefik:
    image: traefik:1.5.4
    restart: always
    command: --api --docker
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    networks:
      - web
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /home/manager/docker/traefik/traefik.toml:/traefik.toml
      - /home/manager/docker/traefik/certs:/certs
    container_name: traefik

networks:
  web:
    external: true

```
traefik.toml
```
checkNewVersion = true
logLevel = "ERROR"
defaultEntryPoints = ["https","http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
    permanent = true
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      certFile = "certs/ec.pem"
      keyFile = "certs/ec.key"

[retry]

[docker]
endpoint = "tcp://127.0.0.1:2377"
domain = "xxxxxx.xx"
watch = true
exposedByDefault = false

[api]
  entryPoint = "traefik"
  dashboard = true
  debug = true

```
If needed the GLPI stack docker-compose file (classic Apache + php-fpm with some scripting for downloading up to date release of GLPI and do some file management)
```

version: '2.5'

services:
  web:
    build: ./web
    container_name: glpiweb
    depends_on:
      - php
    networks:
      - web
      - default
    volumes:
        - ./data:/var/www/html/data
        - glpi_core:/var/www/html/core
        - /etc/timezone:/etc/timezone:ro
        - /etc/localtime:/etc/localtime:ro
    labels:
      - "traefik.docker.network=web"
      - "traefik.enable=true"
      - "traefik.frontend.rule=Host:xxx.xxxxxxxxxxx.xx"
  php:
    build: ./php
    container_name: glpiphp
    volumes:
        - ./data:/var/www/html/data
        - glpi_core:/var/www/html/core
        - /etc/timezone:/etc/timezone:ro
        - /etc/localtime:/etc/localtime:ro
  mysql:
    container_name: glpidb
    image: mariadb:10.3
    volumes:
      - ./base:/var/lib/mysql
    environment:
      - MYSQL_DATABASE=xxxxxxxxx
      - MYSQL_USER=xxxxxxx
      - MYSQL_PASSWORD=xxxxxxxxxxxx
      - MYSQL_ROOT_PASSWORD=xxxxxxxxxxxxx
  cronjobber:
    image: docker:edge
    container_name: glpicron
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    command: >
         sh -c "
         printf '*/1 * * * * docker exec glpiphp /usr/local/bin/php /var/www/html/core/front/cron.php \n' > /etc/crontabs/root
         && printf '*/1 * * * * docker exec glpiphp rsync -a --delete /var/www/html/data/plugins/ /var/www/html/core/plugins/ \n' >> /etc/crontabs/root
         && printf '*/1 * * * * docker exec glpiphp rsync -a --delete /var/www/html/core/marketplace/ /var/www/html/data/marketplace/ \n' >> /etc/crontabs/root
         && crond -f"
         #echo '*/2 * * * * docker exec $$(docker ps -q -f name=GLPI_php) /usr/local/bin/php /var/www/html/core/front/cron.php' > /etc/crontabs/root
volumes:
  glpi_core:
networks:
    web:
        external: true
```

If more information is needed, I can share without any problem.

**Comment by cconard96** [comment]
In this plugin's hook.php file can you change `$_SERVER['HTTPS']` to `$_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https'` on line 25?

There may be other placed it needs changed but it should be enough to verify the fix if it let's you at least install the plugin.

**Comment by ashledombos** [comment]
Thanks, both plugins now can be installed.

**Comment by cconard96** [comment]
That's great news. I'll try to roll out a new version of both plugins with the official fixes later tonight.

**Comment by cconard96** [comment]
Fixed back in v1.0.3 release.
